ffi_internals/src/type_ffi.rs

```diff
@@ -0,0 +1,534 @@
+//!
+//! Contains structures describing the data for a Rust type, and implementations for building the
+//! related FFI and consumer implementations.
+//!
+
+use crate::{
+    parsing,
+    parsing::{FieldAttributes, WrappingType},
+};
+use heck::SnakeCase;
+use proc_macro2::TokenStream;
+use proc_macro_error::{abort, OptionExt};
+use quote::{format_ident, quote};
+use syn::{spanned::Spanned, Ident, Type};
+
+const STRING: &str = "String";
+const STR: &str = "str";
+const DATETIME: &str = "NaiveDateTime";
```

Comment:
Just curious... why map to `NativeDateTime`?

Reply:
Haven't implemented support for generics widely enough (or specifically for chrono) to use `DateTime<UTC>`, so we convert whatever kind of datetime we have to a `chrono::NaiveDateTime` (which is convertible to/from `ffi_core::datetime::TimeStamp` for FFI). It's okay as long as we convert to and expect the timestamps to be in UTC, but it's definitely not ideal. We have this in the backlog for making it better https://agrian.atlassian.net/browse/DEV-13190.